src/services/category_api.js

```diff
@@ -0,0 +1,12 @@
+import {
+    useEffect,
+    useState
+} from "react";
+import axios from "axios";
+
+function useCategory() {
+    const [category, setCategory] = useState([]);
+    useEffect(() => axios.get('/api/categories').then(resp => setCategory(resp.data.categories).catch(err => console.error(err))), []);
```

Comment:
Use a separate file for the api calls.  
Its good practice to move the API calls into the service folder and then get that function into a different file

Reply:
i am actually using a different folder for that!